### PR TITLE
[Cocoa] Delete checks for iOS 14, iOS 15, watchOS 7, watchOS 8, tvOS 14, and tvOS 15

### PR DESCRIPTION
**Pull request by litherum**
#### de0447e0f0d71d86a4db9d9898569ef76085a1d4
<pre>
[Cocoa] Delete checks for iOS 14, iOS 15, watchOS 7, watchOS 8, tvOS 14, and tvOS 15
<a href="https://bugs.webkit.org/show_bug.cgi?id=244829">https://bugs.webkit.org/show_bug.cgi?id=244829</a>

Reviewed by NOBODY (OOPS!).

Mechanical deletion.

No tests because there is no behavior change.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WTF/wtf/PlatformUse.h:
</pre>
https://github.com/WebKit/WebKit/commit/de0447e0f0d71d86a4db9d9898569ef76085a1d4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/88332 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/32737 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/19151 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/97523 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/152993 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/31243 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/26903 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/80533 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92180 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/93940 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/24872 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/75156 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/24851 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/79775 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/79890 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/67817 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/80066 "Built successfully and passed tests") | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/28829 "Hash de0447e0 for PR 4045 does not build (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/13868 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/73827 "Built successfully") | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/28814 "Hash de0447e0 for PR 4045 does not build (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/14882 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26192 "Passed tests") | 
| | [❌ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/31999 "Hash de0447e0 for PR 4045 does not build (failure)") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/37795 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/76679 "Built successfully") | 
| | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/30991 "Hash de0447e0 for PR 4045 does not build (failure)") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34008 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17017 "Passed tests") |

### CODE REVIEW

File: Source/WTF/wtf/PlatformHave.h [hunk 1]
```diff
@@ -407,9 +407,9 @@
 #endif
 
 #if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 110300) \
-    || (((PLATFORM(IOS) && !PLATFORM(IOS_FAMILY_SIMULATOR)) || PLATFORM(MACCATALYST)) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 140500) \
-    || (PLATFORM(WATCHOS) && !PLATFORM(IOS_FAMILY_SIMULATOR) && __WATCH_OS_VERSION_MAX_ALLOWED >= 70400) \
-    || (PLATFORM(APPLETV) && !PLATFORM(IOS_FAMILY_SIMULATOR) && __TV_OS_VERSION_MAX_ALLOWED >= 140500)
+    || (((PLATFORM(IOS) && !PLATFORM(IOS_FAMILY_SIMULATOR)) || PLATFORM(MACCATALYST))) \
```
Comment (litherum) [inline Source/WTF/wtf/PlatformHave.h:410]: `(())`

File: Source/WTF/wtf/PlatformUse.h [hunk 2]
```diff
@@ -301,17 +301,11 @@
 #define USE_CTFONTGETADVANCES_WORKAROUND 1
 #endif
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED < 120000) \
-    || ((PLATFORM(IOS) || PLATFORM(MACCATALYST)) && __IPHONE_OS_VERSION_MIN_REQUIRED < 150000) \
-    || (PLATFORM(WATCHOS) && __WATCH_OS_VERSION_MIN_REQUIRED < 80000) \
-    || (PLATFORM(APPLETV) && __TV_OS_VERSION_MIN_REQUIRED < 150000)
+#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED < 120000)
```
Comment (litherum) [inline Source/WTF/wtf/PlatformUse.h:304]: Unnecessary parentheses

File: Source/WTF/wtf/PlatformUse.h [hunk 3]
```diff
@@ -301,17 +301,11 @@
...
 #define USE_CTFONTSHAPEGLYPHS_WORKAROUND 1
 #endif
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED < 120000) \
-    || ((PLATFORM(IOS) || PLATFORM(MACCATALYST)) && __IPHONE_OS_VERSION_MIN_REQUIRED < 150000) \
-    || (PLATFORM(WATCHOS) && __WATCH_OS_VERSION_MIN_REQUIRED < 80000) \
-    || (PLATFORM(APPLETV) && __TV_OS_VERSION_MIN_REQUIRED < 150000)
+#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED < 120000)
```
Comment (litherum) [inline Source/WTF/wtf/PlatformUse.h:308]: Ditto.

**Review by hortont424 — APPROVED** [review]

**Comment by hortont424** [comment]
(obviously fix whatever the bots are unhappy about)

File: Source/WTF/wtf/PlatformEnableCocoa.h [hunk 4]
```diff
@@ -131,7 +131,7 @@
 
 #endif // !PLATFORM(MAC) && !PLATFORM(MACCATALYST) && !PLATFORM(WATCHOS) && !PLATFORM(APPLETV)
 
-#if !defined(ENABLE_APPLE_PAY_SESSION_V11) && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110300) || (PLATFORM(IOS) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 140500))
+#if !defined(ENABLE_APPLE_PAY_SESSION_V11) && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110300) || PLATFORM(IOS))
```
Comment (aproskuryakov) [inline Source/WTF/wtf/PlatformEnableCocoa.h:134]: __MAC_OS_X_VERSION_MIN_REQUIRED >= 110300 is always true, we don't support targeting 11.0 at this point.

File: Source/WTF/wtf/PlatformHave.h [hunk 5]
```diff
@@ -1311,9 +1311,9 @@
 #endif
 
 #if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 101600) \
```
Comment (aproskuryakov) [inline Source/WTF/wtf/PlatformHave.h:1313]: Don't need this macOS version check.

File: Source/WTF/wtf/PlatformUse.h [hunk 6]
```diff
@@ -323,10 +317,10 @@
 #endif
 
 #if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110000) \
```
Comment (aproskuryakov) [inline Source/WTF/wtf/PlatformUse.h:319]: Don't need this macOS version check.

**Pull request closed without merging**